R/Task.R

```diff
@@ -1250,6 +1273,17 @@ task_check_col_roles.Task = function(task, new_roles, ...) {
     }
   }
 
+  # check offset
+  if (length(new_roles[["offset"]]) && any(fget(task$col_info, new_roles[["offset"]], "type", key = "id") %nin% c("numeric", "integer"))) {
+    stopf("Offset column(s) %s must be a numeric or integer column", paste0("'", new_roles[["offset"]], "'", collapse = ","))
+  }
+
+  if (any(task$missings(cols = new_roles[["offset"]]) > 0)) {
```

Comment:
Maybe use a shorter circuit (something like `mlr3misc::some`)

Reply:
I would use vectorization when available, so the version here is fine i'd say (?)